vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java

```diff
@@ -710,7 +1233,27 @@ private String getStringObject(String propertyName, String object,
      * Clear the list of files being processed, or already uploaded.
      */
     public void clearFileList() {
-        setFiles(Json.createArray());
+        getElement().setPropertyJson("files", Json.createArray());
+    }
+
+    /**
+     * The array of files being processed, or already uploaded.
+     *
+     * @return the {@code files} property from the webcomponent
+     */
+    @Synchronize(property = "files", value = "files-changed")
+    protected JsonArray getFilesJsonArray() {
+        return (JsonArray) getElement().getPropertyRaw("files");
+    }
+
+    /**
+     * Specifies if the maximum number of files have been uploaded
+     *
+     * @return the {@code maxFilesReached} property from the webcomponent
+     */
+    @Synchronize(property = "maxFilesReached", value = "max-files-reached-changed")
+    protected boolean isMaxFilesReachedBoolean() {
```

Comment:
Could be `private` since it's only used internally

Reply:
Done.